--- Eigen/src/Eigenvalues/SelfAdjointEigenSolver.h	2018-12-12 01:57:55.000000000 +0800
+++ Eigen/src/Eigenvalues/SelfAdjointEigenSolver.h	2025-06-30 14:54:04.102782645 +0800
@@ -499,7 +499,7 @@
   {
     for (Index i = start; i<end; ++i)
       if (internal::isMuchSmallerThan(abs(subdiag[i]),(abs(diag[i])+abs(diag[i+1])),precision) || abs(subdiag[i]) <= considerAsZero)
-        subdiag[i] = 0;
+        subdiag[i] = RealScalar(0);
 
     // find the largest unreduced block
     while (end>0 && subdiag[end-1]==RealScalar(0))
