--- Eigen/src/Eigenvalues/Tridiagonalization.h	2021-08-19 04:41:58.000000000 +0800
+++ Eigen/src/Eigenvalues/Tridiagonalization.h	2025-03-29 22:23:43.833858213 +0800
@@ -495,9 +495,9 @@
       subdiag[1] = mat(2,1) - m01 * q;
       if (extractQ)
       {
-        mat << 1,   0,    0,
-               0, m01,  m02,
-               0, m02, -m01;
+        mat << RealScalar(1),   RealScalar(0),    RealScalar(0),
+               RealScalar(0), m01,  m02,
+               RealScalar(0), m02, -m01;
       }
     }
   }
